metadata-io/src/main/java/com/linkedin/metadata/entity/EntityServiceImpl.java

```diff
@@ -1054,43 +871,19 @@ public String batchApplyRetention(Integer start, Integer count, Integer attemptW
     return result.toString();
   }
 
-  private boolean emitChangeLog(@Nullable RecordTemplate oldAspect, @Nullable SystemMetadata oldSystemMetadata,
-      RecordTemplate newAspect, SystemMetadata newSystemMetadata,
-      MetadataChangeProposal mcp, Urn entityUrn,
-      AuditStamp auditStamp, AspectSpec aspectSpec) {
-    boolean isNoOp = oldAspect == newAspect;
-    if (!isNoOp || _alwaysEmitChangeLog || shouldAspectEmitChangeLog(aspectSpec)) {
-      log.debug("Producing MetadataChangeLog for ingested aspect {}, urn {}", mcp.getAspectName(), entityUrn);
-
-      final MetadataChangeLog metadataChangeLog = constructMCL(mcp, urnToEntityName(entityUrn), entityUrn,
-          isNoOp ? ChangeType.RESTATE : ChangeType.UPSERT, aspectSpec.getName(), auditStamp, newAspect, newSystemMetadata,
-          oldAspect, oldSystemMetadata);
-
-      log.debug("Serialized MCL event: {}", metadataChangeLog);
-
-      produceMetadataChangeLog(entityUrn, aspectSpec, metadataChangeLog);
-      preprocessEvent(metadataChangeLog);
-
-      return true;
-    } else {
-      log.debug(
-          "Skipped producing MetadataChangeLog for ingested aspect {}, urn {}. Aspect has not changed.",
-          mcp.getAspectName(), entityUrn);
-      return false;
-    }
-  }
-
-  private void preprocessEvent(MetadataChangeLog metadataChangeLog) {
+  private boolean preprocessEvent(MetadataChangeLog metadataChangeLog) {
     if (_preProcessHooks.isUiEnabled()) {
       if (metadataChangeLog.getSystemMetadata() != null) {
         if (metadataChangeLog.getSystemMetadata().getProperties() != null) {
           if (UI_SOURCE.equals(metadataChangeLog.getSystemMetadata().getProperties().get(APP_SOURCE))) {
             // Pre-process the update indices hook for UI updates to avoid perceived lag from Kafka
             _updateIndicesService.handleChangeEvent(metadataChangeLog);
+            return true;
```

Comment:
Lets report back up the result chain whether we actually ran this. More of an fyi at this point.